host/src/server/api/v1/proof.rs

```diff
@@ -48,15 +46,21 @@ fn set_cached_input(
     network: &str,
     input: &GuestInput,
 ) -> HostResult<()> {
-    if let Some(dir) = cache_path.as_ref() {
-        let path = get_input_path(dir, block_number, network);
-        if !path.exists() {
-            let file = File::create(&path).map_err(<std::io::Error as Into<HostError>>::into)?;
-            info!("caching input for {path:?}");
-            bincode::serialize_into(file, input).map_err(|e| HostError::Anyhow(e.into()))?;
-        }
+    let Some(dir) = cache_path.as_ref() else {
+        return Ok(());
+    };
+
+    let path = get_input_path(dir, block_number, network);
+
+    if path.exists() {
+        return Ok(());
     }
-    Ok(())
+
+    let file = File::create(&path).map_err(<std::io::Error as Into<HostError>>::into)?;
+
+    info!("caching input for {path:?}");
+
+    bincode::serialize_into(file, input).map_err(|e| HostError::Anyhow(e.into()))
```

Comment:
For this one I kind of prefer the original version because less tricky flow, though both are fine.